### DISCUSSION TITLE
Create mutiple blobs in one request

### DISCUSSION THREAD

**Discussion by Akrabut** [octokit/octokit.js#2241] [General]
**My use case**:
I want to commit many changes to a Github repo I control programatically by generating blobs, then generating a tree out of it, then creating a commit out of it and then a PR or a commit to a PR.

**What I do**:
I'm using the `octokit.js` package.
I use the `createBlob` for each change.
I use the created blobs to `createTree`.
I use the created tree to commit.

**The issue**:
There doesn't seem to be a way to create blobs in batches, so I have to send several request per single file change, which results in Github (implicitly, mind you) rate limiting me.

Is there any way to create blobs in batches? if so, where is it documented?
If not, how does a remote push to Github works in case of large amount of changes?

**Comment by gr2m** [comment]
I think you are looking for this plugin:
https://github.com/gr2m/octokit-plugin-create-pull-request

**Reply by Akrabut** [reply]
I might've forgotten to mention, but the many file changes/creations I commit contains images, so if I understand correctly, using this plugin [would result in the same behavior](https://github.com/gr2m/octokit-plugin-create-pull-request/blob/28f33b6daabb219d914f6993c96175c821e3719c/src/create-tree.ts#L78) (note - the issue is getting rate limited due to too many blob creation requests).

**Reply by reduckted** [reply]
@Akrabut Did you end up finding a solution for this? I'm running into the same problem when creating commits with lots of files. Creating the blobs results in being rate limited. 😢

**Comment by Akrabut** [comment]
It has been a while so I'm not sure, but I believe I sent the changes in
batches rather than all at once.

בתאריך יום ה׳, 5 ביוני 2025, 03:08, מאת Dave ***@***.***>:
